msal/src/main/java/com/microsoft/identity/client/statemachine/Error.kt

```diff
@@ -0,0 +1,99 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.client.statemachine
+
+sealed class Error(
```

Comment:
Are we always sending back Error objects for error in Native Auth or are we still throwing Exceptions in certain cases? Either way, why?

Reply:
we are sending back error objects. @SammyO  Can you provide the reason for this?

Reply:
We're doing both. Any expected error will be returned as an Error object (e.g. PasswordInvalid, GeneralError). If an exception happens in one of the underlying SDK layers, it might also be thrown back to the developer. I.e. we don't have a try-catch in the SDK interface that catches all exceptions and turns them in to Errors.

Reply:
This looks like putting too much work on the developers. They have to be on the look out for both Errors and Exceptions. I'll rather prefer having to worry about only one of these but its up to you.